Telegram notifications, send email using server requiring authentication and SSL

velika12 opened the pull request.

mylk commented:
Hello @velika12!

Thanks for contributing, nice additions!

Can you also please add the new options to the example.ini file (configuration file) too?
Also add a bullet on README about the support for Telegram, there is a section about the notification types.

Thanks!

velika12 commented:
Hello, @mylk!

I've added what you asked for. I've also added some tests for the new functionality and found a bug corresponding to `ConfigParser.ConfigParser` constructor in tests. The deal was that not all tests in `test_optionresolver.py` require it mocked, but once mocked this method is not reassigned to the default logic later. So as all tests run in random order, there appears unnecessary mock object when we need the default one.

https://stackoverflow.com/questions/11746431/any-way-to-reset-a-mocked-method-to-its-original-state-python-mock-mock-1-0

mylk commented:
Yes, you are right.
I think Unittest executes the tests in alphabetical order :-P

We use that trick in other others test too to overcome the accidental mock usage.

In general, I'm preparing a commit for setup_options, to make it more flexible.
